### PR TITLE
unified dockerfile that supports multi arch

### PR DESCRIPTION
**Pull request by hriprsd**
gh workflow has been updated as well - with proper build args

### CODE REVIEW

**Comment by hriprsd** [comment]
solves #2585

File: Dockerfile [hunk 1]
```diff
@@ -16,23 +18,40 @@ RUN dotnet restore NineChronicles.Headless.Executable
...
+    --version-suffix $COMMIT
+else
+  echo "Not supported target platform: '$TARGETPLATFORM'."
+  exit -1
+fi
+EOF
 
 # Build runtime image
-FROM mcr.microsoft.com/dotnet/aspnet:8.0
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/aspnet:8.0-bookworm-slim
```
Comment (moreal) [inline Dockerfile:47]: Note: according to the [dotnet/dotnet-docker repo](https://github.com/dotnet/dotnet-docker/blob/main/README.aspnet.md#linux-amd64-tags), `8.0-bookworm-slim` is same with the `8.0`

Comment (moreal) [inline Dockerfile:47]: What does it use `BUILDPLATFORM`? Shouldn't it use `TARGETPLATFORM` instead?

https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

**Comment by hriprsd** [comment]
updated code as per review comments. Please check - thanks!

**Review by moreal — APPROVED** [review]
Thank you for your first contribution! Thanks to your contribution, we would become able to build Docker image with single Dockerfile!! 🚀 

---------

We're now operating Modathon 2024 (Nine Chronicles Hackathon) with a prize pool about $10,000. If you're interested, please take a look at the following items 🙏🏻 
https://dorahacks.io/hackathon/planetarium-2024modathon/detail
https://planetarium.dev/discord

File: .github/workflows/push_docker_image.yml [hunk 2]
```diff
@@ -54,13 +54,15 @@ jobs:
             -f ${{ matrix.docker.dockerfile }}.amd64 \
             -t ${{ matrix.docker.repo }}:git-${{ github.sha }}-amd64 \
             --build-arg COMMIT=git-${{ github.sha }}
+            --build-arg TARGETPLATFORM=linux/amd64
```
Comment (moreal) [inline .github/workflows/push_docker_image.yml:57]: Oops could you add backslash to treat a new line as an option of the `docker build` command?
```suggestion
            --build-arg COMMIT=git-${{ github.sha }} \
            --build-arg TARGETPLATFORM=linux/amd64
```

File: .github/workflows/push_docker_image.yml [hunk 3]
```diff
@@ -54,13 +54,15 @@ jobs:
...
           docker push ${{ matrix.docker.repo }}:git-${{ github.sha }}-amd64
       - name: build-and-push-arm64v8
         run: |
           docker build . \
             -f ${{ matrix.docker.dockerfile }}.arm64v8 \
             -t ${{ matrix.docker.repo }}:git-${{ github.sha }}-arm64v8 \
             --build-arg COMMIT=git-${{ github.sha }}
+            --build-arg TARGETPLATFORM=linux/arm64
```
Comment (moreal) [inline .github/workflows/push_docker_image.yml:65]:
```suggestion
            --build-arg COMMIT=git-${{ github.sha }} \
            --build-arg TARGETPLATFORM=linux/arm64
```

Comment (hriprsd) [reply .github/workflows/push_docker_image.yml]: Good catch, my bad. Fixing now!

**Comment by hriprsd** [comment]
fails on install dependencies step 😢

**Comment by moreal** [comment]
> fails on install dependencies step 😢

@hriprsd Ah that's a problem with the Libplanet dependency. Looks like we'll have to fix that separately.

**Comment by moreal** [comment]
It will be resolved by using https://github.com/planetarium/libplanet/releases/tag/5.3.1 and I have opened a new PR https://github.com/planetarium/lib9c/pull/2898 to apply it to Lib9c. After the PR is merged, I'll bump the Lib9c submodule and this PR will be resolved as well.

**Comment by hriprsd** [comment]
> It will be resolved by using https://github.com/planetarium/libplanet/releases/tag/5.3.1 and I have opened a new PR [planetarium/lib9c#2898](https://github.com/planetarium/lib9c/pull/2898) to apply it to Lib9c. After the PR is merged, I'll bump the Lib9c submodule and this PR will be resolved as well.

Thanks , hoping this will be done by October end 🤞 #hacktoberfest

**Comment by hriprsd** [comment]
Also, could you please add `hacktoberfest-accepted` label to my PR please? would be a huge help! Thanks in advance!

**Comment by moreal** [comment]
A PR that fixes the original issue has been merged into the development branch - this PR will probably succeed soon!

**Pull request merged**